debezium-server/debezium-server-redis/src/main/java/io/debezium/server/redis/RedisDatabaseHistory.java

```diff
@@ -0,0 +1,222 @@
+            Uni.createFrom().item(() -> {
+                return (StreamEntryID) client.xadd(this.redisKeyName, (StreamEntryID) null, Collections.singletonMap("schema", line));
+            })
+                    .onFailure().invoke(
+                            f -> {
+                                LOGGER.warn("Writing to database history stream failed with " + f);
+                                LOGGER.warn("Will retry");
+                            })
+                    .onFailure(JedisConnectionException.class).invoke(
+                            f -> {
+                                LOGGER.warn("Attempting to reconnect to redis ");
+                                this.connect();
+                            })
+                    // retry on failure with backoff
+                    .onFailure().retry().withBackOff(Duration.ofMillis(initialRetryDelay), Duration.ofMillis(maxRetryDelay)).indefinitely()
+                    // write success trace message
+                    .invoke(
+                            item -> {
+                                LOGGER.trace("Record written to database history in redis: " + line);
+                            })
+                    .await().indefinitely();
+        }
+        catch (IOException e) {
+            logger.error("Failed to convert record to string: {}", record, e);
+        }
+    }
+
+    @Override
+    public void stop() {
+        running.set(false);
+        super.stop();
+    }
+
+    @Override
+    protected synchronized void recoverRecords(Consumer<HistoryRecord> records) {
+        List<StreamEntry> entries = Uni.createFrom().item(
+                () -> {
+                    return (List<StreamEntry>) client.xrange(
+                            this.redisKeyName, (StreamEntryID) null, (StreamEntryID) null);
+                })
+                .onFailure().invoke(
+                        f -> {
+                            LOGGER.warn("Reading from database history stream failed with " + f);
+                            LOGGER.warn("Will retry");
+                        })
+                .onFailure(JedisConnectionException.class).invoke(
+                        f -> {
+                            LOGGER.warn("Attempting to reconnect to redis ");
+                            this.connect();
+                        })
+                // retry on failure with backoff
+                .onFailure().retry().withBackOff(Duration.ofMillis(initialRetryDelay), Duration.ofMillis(maxRetryDelay)).indefinitely()
+                .await().indefinitely();
+        for (StreamEntry item : entries) {
+            try {
+                records.accept(new HistoryRecord(reader.read(item.getFields().get("schema"))));
+            }
+            catch (IOException e) {
+                logger.error("Failed to convert record to string: {}", item, e);
+            }
+        }
+
+    }
+
+    @Override
+    public boolean storageExists() {
+        return true;
```

Comment:
So this is usually used (together with `initializeStorage()`) for setting up the "storage", i.e. creating a topic, file, etc. Is that not needed for Redis?

Reply:
no. the data structure (Stream in this case) is set up automatically the first time the entry is written. It is always the equivalent of "auto create topic" in Kafka

Reply:
Ok, that's what I thought, thanks for confirming.